interfaces/cython/cantera/onedim.pyx

```diff
@@ -837,37 +837,51 @@ cdef class Sim1D:
         nPoints = [len(flow_domains[0].grid), 12, 24, 48]
 
         for N in nPoints:
+            if N > self.get_max_grid_points():
+                raise CanteraError('Maximum number of grid points exceeded')
+
             for i,D in enumerate(flow_domains):
                 if N != len(D.grid):
                     D.grid = np.linspace(zmin[i], zmax[i], N)
 
             self.set_initial_guess(*self._initial_guess_args,
                                    **self._initial_guess_kwargs)
 
+            # Try solving with energy enabled, which usually works
+            log('Solving on {} point grid with energy equation enabled', N)
+            self.energy_enabled = True
             try:
-                # Try solving with energy enabled, which usually works
-                log('Solving on {} point grid with energy equation enabled', N)
-                self.energy_enabled = True
                 self.sim.solve(loglevel, <cbool>False)
-                solved = True
             except CanteraError as e:
                 log(str(e))
                 solved = False
+            else:
```

Comment:
The other refactoring here is fine, but I would much rather see `solved = True` in the `try` block than use the `try...else` construct (which I have to admit I find confusing).